de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/ABeanStructuralElementInstance.java

```diff
@@ -109,6 +129,34 @@ public Command remove(EditingDomain ed, IBeanCategoryAssignment bca) {
 		return bcaHelper.getAllBeanCategories(sei, catBeanClazz);
 	}
 	
+	@Override
+	public List<BeanCategoryAssignment> getCategoryAssignments() {
+		ArrayList<BeanCategoryAssignment> beans = new ArrayList<BeanCategoryAssignment>();
+		
+		for (CategoryAssignment ca : sei.getCategoryAssignments()) {
+			BeanCategoryAssignment bean = new BeanCategoryAssignment();
+			bean.setTypeInstance(ca);
+			beans.add(bean);
+		}
+		
+		return beans;

```

Comment:
We should be aware, that the list we are handing back, does not allow to directly add a CA to the SEI, since it is realy just a pure List of CAs. I think it is ok in case that we are clear on the API contract.

Do we want to have it like this, since operations such as Bean.getCategoryAssignments().add(CA) will not add a CA to the Bean.

Reply:
After discussion with @SaMuellerDLR we think the returned list should be wrapped to an unmodifiable collection if posible. This would be in line with the contract of all the other metods to get all CAs which are internalley calling the BeanStructuralElementInstanceHelper.

Reply:
That makes sense I will wrapp it into an unmodifiable collection

Reply:
Now JAXB doesn't use the functions of the interface, but instead some private functions shadowing the behavior. This way the original functions can return unmodifiable lists.